cmd/clone.go

```diff
@@ -1294,7 +1314,7 @@ func pruneRepos(cloneTargets []scm.Repo) int {
 	count := 0
 	colorlog.PrintInfo("\nScanning for local clones that have been removed on remote...")
 
-	files, err := os.ReadDir(outputDirAbsolutePath)
+	files, err := readDirRecursively(outputDirAbsolutePath)
```

Comment:
To reduce the chances of hitting strange edge cases lets only call this when the `--preserve-dir` is set, otherwise lets default to current implementation.